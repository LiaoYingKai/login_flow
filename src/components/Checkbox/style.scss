.checkbox {
	display: inline-block;
	label {
		cursor: pointer;
		display: inline-block;
		width: 16px;
		height: 16px;
		background: #fff;
		border-radius: 3px;
		text-align: center;
	}
	input[type="checkbox"] {
		display: none;
		&:checked ~ label {
			&:before {
				display: inline-block;
				content: "âœ“";
				line-height: 16px;
				color: #358ca7;
			}
		}
	}
}