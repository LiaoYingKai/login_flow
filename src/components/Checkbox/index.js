import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './style.scss';

const propTypes = {
	onChange: PropTypes.func,
	className: PropTypes.string,
	value: PropTypes.bool,
};

const defaultProps = {
	onChange: () => {},
};

function Checkbox({
	value,
	onChange,
	className
}) {
	return (
		<div className={cx("checkbox", className)}>
			<input
				type="checkbox"
				id="checkbox"
				onChange={onChange}
				value={value}
			/>
			<label htmlFor="checkbox"></label>
		</div>
	);
}

Checkbox.propTypes = propTypes;
Checkbox.defaultProps = defaultProps;

export default Checkbox;
