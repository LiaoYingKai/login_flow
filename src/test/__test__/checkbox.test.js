/* eslint-disable no-undef */
import React from 'react';
import Checkbox from '../../components/Checkbox';
import { shallow, mount, } from "enzyme";
import toJson from 'enzyme-to-json';

describe('Checkbox', () => {
	it('should handle default props', () => {
		const {
			onChange
		} = Checkbox.defaultProps;

		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should select by checkbox class', () => {
		const wrapper = shallow(<Checkbox/>);

		expect(wrapper.hasClass('checkbox')).toEqual(true);
	});

	it('should select by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<Checkbox className={className}/>);

		expect(wrapper.hasClass('mock-class')).toEqual(true);
	});

	it('sholuld handle onChange callback', () => {
		const onChange = jest.fn();
		const wrapper = shallow(<Checkbox onChange={onChange} />);

		wrapper.find('input').at(0).simulate('change', { target: { checked: true } });
		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenCalled();
	});

	it('should render corrent', () => {
		const onChange = () => {};
		const value = false;
		const wrapper = shallow(<Checkbox onChange={onChange} value={value}/>);

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-class';
		const onChange = () => {};
		const value = false;
		const wrapper = mount(<Checkbox onChange={onChange} className={className} value={value}/>);

		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().onChange).toEqual(onChange);
		expect(wrapper.props().value).toEqual(value);
	});
});
