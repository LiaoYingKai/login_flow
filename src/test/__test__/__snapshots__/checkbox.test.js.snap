// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render corrent 1`] = `
<div
  className="checkbox"
>
  <input
    id="checkbox"
    onChange={[Function]}
    type="checkbox"
    value={false}
  />
  <label
    htmlFor="checkbox"
  />
</div>
`;
